flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -897,7 +897,7 @@ class TaskManager(
         config.timeout,
         libCache,
         fileCache,
-        runtimeInfo)
+        new TaskRuntimeInfo(hostname, taskManagerConfig, tdd.getAttemptNumber))
```

Comment:
Ah. Yes. That makes sense. I will revert this and open a separate PR. Apologies.